src/app/homepage/ResearchThemesSection.tsx

```diff
@@ -43,47 +45,65 @@ const ResearchTopicsMemberAvatar = styled(Image)`
   height: auto;
 `
 
+const GatherStatsByResearchTopic = () => {
+  // TODO: Somehow avoid having to define the keys explicitly. Typescript doesn't allow empty definition
+  const statsByResearchTopic: Record<ResearchTopicType, { numPublications: number; authors: IMember[] }> = {} as any
+  ResearchTopics.forEach(topic => {
+    const filteredPublications = PUBLICATIONS.filter(publication => publication.topics.includes(topic))
+    const numPublications = filteredPublications.length
+    const topicAuthors = filteredPublications
+      .sort((a, b) => b.year - a.year)
+      .flatMap(publication => publication.authors)
+    const filteredAuthors = Object.values(MEMBERS).filter(member =>
+      topicAuthors.includes(`${member.firstName} ${member.lastName}`)
+    )
+    statsByResearchTopic[topic] = { numPublications, authors: filteredAuthors }
+  })
+  return statsByResearchTopic
+}
+
 export const ResearchThemesSection = () => {
-  const [membersByResearchTopic, numVisible] = useMemo(() => {
+  const [statsByResearchTopic, numVisible] = useMemo(() => {
     // TODO: Specify the type of membersByResearchTopic
-    const membersByResearchTopic: Record<string, any[]> = {}
-    ResearchTopics.forEach(topic => {
-      membersByResearchTopic[topic] = MEMBERS.filter(
-        member => member.researchTopics && member.researchTopics.includes(topic)
-      )
-    })
-    return [membersByResearchTopic, 4]
+    const statsByResearchTopic: Record<string, any> = GatherStatsByResearchTopic()
```

Comment:
isn't the return value always the same? If we call the function inside the component, the function is called everytime the component is rendered. Use either useMemo or compute the value outside of the component.

Reply:
The entire thing is wrapped inside `useMemo` so it should only run once on mount, no?